pyazhpc/azhpc.py

```diff
@@ -542,73 +551,62 @@ def do_slurm_resume(args):
 
     log.info(f"slurm resume for {args.nodes}")
     # first get the resource name
-    resource_name, brackets = re.search(r'([^[]*)\[?([\d\-\,]*)\]?', args.nodes).groups(0)
-    resource_list = []
-    if bool(brackets):
-        for part in brackets.split(","):
-            if "-" in part:
-                lo, hi = part.split("-")
-                assert len(lo) == 4, "expecting number width of 4"
-                assert len(hi) == 4, "expecting number width of 4"
-                for i in range(int(lo), int(hi) + 1):
-                    resource_list.append(f"{resource_name}{i:04d}")
-            else:
-                assert len(part) == 4, "expecting number width of 4"
-                resource_list.append(f"{resource_name}{part}")
-    else:
-        resource_list.append(resource_name)
-        resource_name = resource_name[:-4]
-    
-    template_resource = config.get("resources", {}).get(resource_name)
-    if not template_resource:
-        log.error(f"{resource_name} resource not found in config")
-        sys.exit(1)
-    if template_resource.get("type") != "slurm_partition":
-        log.error(f"invalid resource type for scaling")
-    
-    template_resource["type"] = "vm"
-    del template_resource["instances"]
-
-    log.info(f"resource_name= {resource_name}")
-    log.info("resource_list= " + ",".join(resource_list))
-    
-    config["resources"] = {}
-    for rname in resource_list:
-        config["resources"][rname] = template_resource
+    resource_names, resource_list = _nodelist_expand(args.nodes)
 
-    tpl = arm.ArmTemplate()
-    tpl.read_resources(config, False)
+    # Loop over all resources
+    for resource in resource_names:
+        template_resource = config.get("resources", {}).get(resource)
+        if not template_resource:
+            log.error(f"{resource} resource not found in config")
+            sys.exit(1)
+        if template_resource.get("type") != "slurm_partition":
+            log.error(f"invalid resource type for scaling")
 
-    output_template = f"deploy_{args.config_file}_{timestamp}"
+        template_resource["type"] = "vm"
+        del template_resource["instances"]
 
-    log.info("writing out arm template to " + output_template)
-    with open(output_template, "w") as f:
-        f.write(tpl.to_json())
+        log.info(f"resource= {resource}")
+        log.info("resource_list= " + ",".join(resource_list))
 
-    log.info("deploying arm template")
-    deployname = azutil.deploy(
-        config["resource_group"],
-        output_template
-    )
-    log.debug(f"deployment name: {deployname}")
+        config["resources"] = {}
 
-    _wait_for_deployment(config["resource_group"], deployname)
+        # Iterate over all nodes which name starts with the resource name
+        for rname in filter(lambda x: x.startswith(resource), resource_list):
```

Comment:
this would fail if you had two resource names where one is a substring of the other, e.g. viz and viz3d (which is currently the case for AzureHPC OnDemand)

Reply:
Good point. Let me make it more robust.

Reply:
@edwardsp This is not an issue in the Slurm setup deployed here since each partition contains nodes belonging to the same resource and jobs cannot be submitted across multiple partitions.
That said, it remains a good point. I have modified the logic to correctly handle multiple resources in the same resume in case users manually add a partition that includes nodes belonging to multiple resources.